src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1447,50 +1450,66 @@ def update_site_appsettings(stack, site_config):
-            for fx in java_container_stack['properties']['frameworks']:
-                for fx_version in fx['majorVersions']:
+        if self._linux:
+            for name, properties in [(s['name'], s['properties']) for s in stacks]:
+                for major in properties['majorVersions']:
+                    default_minor = next((m for m in (major['minorVersions'] or []) if m['isDefault']),
+                                         None)
                     result.append({
-                        'displayName': 'java|{}|{}|{}'.format(java_version['displayVersion'],
-                                                              fx['display'],
-                                                              fx_version['displayVersion']),
+                        'displayName': (default_minor['runtimeVersion']
+                                                    if default_minor else major['runtimeVersion'])
+                    })
+        else: #Windows stacks
+            config_mappings = {
+                'node': 'WEBSITE_NODE_DEFAULT_VERSION',
+                'python': 'python_version',
+                'php': 'php_version',
+                'aspnet': 'net_framework_version'
+            }
+            
+            
+            # get all stack version except 'java'
+            for name, properties in [(s['name'], s['properties']) for s in stacks
+                                     if s['name'] in config_mappings]:
+                for major in properties['majorVersions']:
+                    default_minor = next((m for m in (major['minorVersions'] or []) if m['isDefault']),
+                                         None)
+                    result.append({
+                        'displayName': name + '|' + major['displayVersion'],
                         'configs': {
-                            'java_version': java_version['runtimeVersion'],
-                            'java_container': fx['name'],
-                            'java_container_version': fx_version['runtimeVersion']
+                            config_mappings[name]: (default_minor['runtimeVersion']
+                                                    if default_minor else major['runtimeVersion'])
                         }
                     })
-
-        for r in result:
-            r['setter'] = (_StackRuntimeHelper.update_site_appsettings if 'node' in
-                           r['displayName'] else _StackRuntimeHelper.update_site_config)
+            
+            # deal with java, which pairs with java container version
+            java_stack = next((s for s in stacks if s['name'] == 'java'))
+            java_container_stack = next((s for s in stacks if s['name'] == 'javaContainers'))
+            for java_version in java_stack['properties']['majorVersions']:
+                for fx in java_container_stack['properties']['frameworks']:
+                    for fx_version in fx['majorVersions']:
```

Comment:
To simplify the embedded loops, you may consider using the [itertools](https://docs.python.org/3/library/itertools.html?highlight=itertools) module.